src/Microsoft.ML.PipelineInference/PipelinePattern.cs

```diff
@@ -183,34 +196,58 @@ public Experiment CreateTrainTestExperiment(IDataView trainData, IDataView testD
         /// <summary>
         /// Runs a train-test experiment on the current pipeline, through entrypoints.
         /// </summary>
-        public double RunTrainTestExperiment(IDataView trainData, IDataView testData, AutoInference.SupportedMetric metric, MacroUtils.TrainerKinds trainerKind)
+        public void RunTrainTestExperiment(IDataView trainData, IDataView testData, 
+            AutoInference.SupportedMetric metric, MacroUtils.TrainerKinds trainerKind, out double testMetricValue,
+            out double trainMetricValue)
         {
-            var experiment = CreateTrainTestExperiment(trainData, testData, trainerKind, out var trainTestOutput);
+            var experiment = CreateTrainTestExperiment(trainData, testData, trainerKind, true, out var trainTestOutput);
             experiment.Run();
+
             var dataOut = experiment.GetOutput(trainTestOutput.OverallMetrics);
             var schema = dataOut.Schema;
             schema.TryGetColumnIndex(metric.Name, out var metricCol);
+            double metricValue = 0;
+            double trainingMetricValue = 0;
 
             using (var cursor = dataOut.GetRowCursor(col => col == metricCol))
             {
                 var getter = cursor.GetGetter<double>(metricCol);
-                double metricValue = 0;
                 cursor.MoveNext();
                 getter(ref metricValue);
-                return metricValue;
+            }
+
+            dataOut = experiment.GetOutput(trainTestOutput.TrainingOverallMetrics);
+            schema = dataOut.Schema;
+            schema.TryGetColumnIndex(metric.Name, out metricCol);
+
+            using (var cursor = dataOut.GetRowCursor(col => col == metricCol))
+            {
+                var getter = cursor.GetGetter<double>(metricCol);
+                cursor.MoveNext();
```

Comment:
>cursor.MoveNext(); [](start = 16, length = 18)

We should validate that this works. We should also validate that the next call to `MoveNext` will return false.

Reply:
Made changes, will push new version.